### PR TITLE
fix(builder): should not apply babel in rspack mode when tools.babel not modify config

### PR DESCRIPTION
**Pull request by 9aoy**
### <samp>🤖 Generated by Copilot at 8caacfe</samp>

This pull request fixes a bug in the `builderPluginBabel` function that caused the babel-loader to be always used in the webpack config, even if the user did not provide any babel options. It also adds tests and a changeset file to document and verify the fix.

### <samp>🤖 Generated by Copilot at 8caacfe</samp>

*  Add markdown file for changeset tool ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-511003e83e69a509b71133e9100b673d3b91f7e0dae422c4774588d72ed04c79R1-R7))
*  Fix bug where babel-loader was always applied even if user did not specify babel options ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-42b824b393afa60dd38acf7a5f7b290dd6ce10a0b5bc08b8ad937cf3c741ba65R7), [link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-42b824b393afa60dd38acf7a5f7b290dd6ce10a0b5bc08b8ad937cf3c741ba65L47-R74), [link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-42b824b393afa60dd38acf7a5f7b290dd6ce10a0b5bc08b8ad937cf3c741ba65L73-R93))
   *  Import `cloneDeep` and `isEqual` functions from `@modern-js/utils/lodash` ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-42b824b393afa60dd38acf7a5f7b290dd6ce10a0b5bc08b8ad937cf3c741ba65R7))
   *  Clone and compare base and user babel configurations in `getBabelOptions` function ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-42b824b393afa60dd38acf7a5f7b290dd6ce10a0b5bc08b8ad937cf3c741ba65L47-R74))
   *  Return empty object if user babel configuration has not modified base configuration ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-42b824b393afa60dd38acf7a5f7b290dd6ce10a0b5bc08b8ad937cf3c741ba65L47-R74))
   *  Destructure and check `includes`, `excludes`, and `babelOptions` properties before setting babel-loader in `builderPluginBabel` function ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-42b824b393afa60dd38acf7a5f7b290dd6ce10a0b5bc08b8ad937cf3c741ba65L73-R93))
*  Add test cases for `builderPluginBabel` function when user babel configuration is null or returns null ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-20e3614ef9f0faff4cd0b7752c7a2ff09b6fde8f0998f5667944b5d7d6468ef9R11-R56))
   *  Use `createBuilder` and `inspectConfig` functions to generate and compare webpack config ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-20e3614ef9f0faff4cd0b7752c7a2ff09b6fde8f0998f5667944b5d7d6468ef9R11-R56))
   *  Use `toMatchSnapshot` matcher to verify expected output ([link](https://github.com/web-infra-dev/modern.js/pull/4648/files?diff=unified&w=0#diff-20e3614ef9f0faff4cd0b7752c7a2ff09b6fde8f0998f5667944b5d7d6468ef9R11-R56))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/builder/builder-rspack-provider/tests/plugins/babel.test.ts [hunk 1]
```diff
@@ -4,6 +4,52 @@ import { builderPluginBabel } from '@/plugins/babel';
 
 describe('plugins/babel', () => {
   it('should set babel-loader', async () => {
+    const builder = await createBuilder({
+      plugins: [builderPluginBabel()],
+      builderConfig: {
+        tools: {
+          babel(config: any) {
+            // 添加一个插件，比如配置某个组件库的按需引入
```
Comment (chenjiahan) [inline packages/builder/builder-rspack-provider/tests/plugins/babel.test.ts:12]: english first

**Review by chenjiahan — APPROVED** [review]
